xcm/xcm-builder/src/integration_tests.rs

```diff
@@ -0,0 +1,413 @@
+		Balances: pallet_balances::{Pallet, Call, Storage, Config<T>, Event<T>},
+		ParasOrigin: origin::{Pallet, Origin},
+		ParasUmp: ump::{Pallet, Call, Storage, Event},
+		XcmPallet: pallet_xcm::{Pallet, Call, Storage, Event<T>},
+	}
+);
+
+pub const ALICE: AccountId = AccountId::new([0u8; 32]);
+pub const PARA_ID: u32 = 2000;
+pub const INITIAL_BALANCE: u128 = 100_000_000_000;
+
+pub fn kusama_like_with_balances(balances: Vec<(AccountId, Balance)>) -> sp_io::TestExternalities {
+	let mut t = frame_system::GenesisConfig::default().build_storage::<Runtime>().unwrap();
+
+	pallet_balances::GenesisConfig::<Runtime> {
+		balances,
+	}
+	.assimilate_storage(&mut t)
+	.unwrap();
+
+	// use polkadot_primitives::v1::{MAX_CODE_SIZE, MAX_POV_SIZE};
+	// // default parachains host configuration from polkadot's `chain_spec.rs`
+	// kusama::ParachainsConfigurationConfig {
+	// 	config: polkadot_runtime_parachains::configuration::HostConfiguration {
+	// 		validation_upgrade_frequency: 1u32,
+	// 		validation_upgrade_delay: 1,
+	// 		code_retention_period: 1200,
+	// 		max_code_size: MAX_CODE_SIZE,
+	// 		max_pov_size: MAX_POV_SIZE,
+	// 		max_head_data_size: 32 * 1024,
+	// 		group_rotation_frequency: 20,
+	// 		chain_availability_period: 4,
+	// 		thread_availability_period: 4,
+	// 		max_upward_queue_count: 8,
+	// 		max_upward_queue_size: 1024 * 1024,
+	// 		max_downward_message_size: 1024,
+	// 		// this is approximatelly 4ms.
+	// 		//
+	// 		// Same as `4 * frame_support::weights::WEIGHT_PER_MILLIS`. We don't bother with
+	// 		// an import since that's a made up number and should be replaced with a constant
+	// 		// obtained by benchmarking anyway.
+	// 		ump_service_total_weight: 4 * 1_000_000_000,
+	// 		max_upward_message_size: 1024 * 1024,
+	// 		max_upward_message_num_per_candidate: 5,
+	// 		hrmp_open_request_ttl: 5,
+	// 		hrmp_sender_deposit: 0,
+	// 		hrmp_recipient_deposit: 0,
+	// 		hrmp_channel_max_capacity: 8,
+	// 		hrmp_channel_max_total_size: 8 * 1024,
+	// 		hrmp_max_parachain_inbound_channels: 4,
+	// 		hrmp_max_parathread_inbound_channels: 4,
+	// 		hrmp_channel_max_message_size: 1024 * 1024,
+	// 		hrmp_max_parachain_outbound_channels: 4,
+	// 		hrmp_max_parathread_outbound_channels: 4,
+	// 		hrmp_max_message_num_per_candidate: 5,
+	// 		dispute_period: 6,
+	// 		no_show_slots: 2,
+	// 		n_delay_tranches: 25,
+	// 		needed_approvals: 2,
+	// 		relay_vrf_modulo_samples: 2,
+	// 		zeroth_delay_tranche_width: 0,
+	// 		..Default::default()
+	// 	},
+	// }.assimilate_storage(&mut t).unwrap();
+
+	let mut ext = sp_io::TestExternalities::new(t);
+	ext.execute_with(|| System::set_block_number(1));
+	ext
+}
+
+fn buy_execution<C>(debt: Weight) -> Order<C> {
+	use xcm::opaque::v0::prelude::*;
+	Order::BuyExecution {
+		fees: All,
+		weight: 0,
+		debt,
+		halt_on_error: false,
+		xcm: vec![],
+	}
+}
+
+#[test]
+fn withdraw_and_deposit_works() {
+	use xcm::opaque::v0::prelude::*;
```

Comment:
this is in all tests so just move out of tests

Reply:
actually, if you do that, you get this error 

```
error[E0308]: `match` arms have incompatible types
  --> xcm/pallet-xcm/src/mock.rs:25:1
   |
25 | / construct_runtime!(
26 | |     pub enum Test where
27 | |         Block = Block,
28 | |         NodeBlock = Block,
...  |
35 | |     }
36 | | );
   | |  ^
   | |  |
   | |  expected enum `std::option::Option`, found enum `xcm::v0::MultiAsset`
   | |__this is found to be of type `std::option::Option<&pallet_balances::Call<Test>>`
   |    `match` arms have incompatible types
   |
   = note: expected enum `std::option::Option<&pallet_balances::Call<Test>>`
              found enum `xcm::v0::MultiAsset`
   = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)
```

Reply:
yeah, `construct_runtime!` collides with the prelude import